ports/stm32/boards/ARDUINO_GIGA/board_init.c

```diff
@@ -35,6 +35,24 @@ void GIGA_board_startup(void) {
 void GIGA_board_early_init(void) {
     HAL_InitTick(0);
 
+    // The following pins need to be configured at this point to fix an
+    // issue with some boards that have an older QSPI flash part revision.
+    GPIO_InitTypeDef GPIO_InitStruct;
+    GPIO_InitStruct.Mode = GPIO_MODE_ANALOG;
+    GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_LOW;
+    GPIO_InitStruct.Pull = GPIO_NOPULL;
+    GPIO_InitStruct.Alternate = 0;
+
+    GPIO_InitStruct.Pin = GPIO_PIN_13;
+    __HAL_RCC_GPIOD_CLK_ENABLE();
+    HAL_GPIO_Init(GPIOD, &GPIO_InitStruct);
```

Comment:
Could it use `mp_hal_pin_config(..., MP_HAL_PIN_MODE_ANALOG, MP_HAL_PIN_PULL_NONE, 0)` here?

Reply:
@dpgeorge It probably can, but this has been tested on one of the boards the have the issue, I don't have one to retest it, but I will ask if someone can retest it.

Reply:
Updated and retested.

Reply:
Thanks for updating, it looks a lot cleaner now.

But, I think the pins should be changed to use the CPU pin names, ie `pin_D13` and `pin_F7`.  I checked, and that change makes no difference to the firmware at all (it's the same binary), so no need to retest that change.

Reply:
Updated.

Reply:
I think I made a mistake here by using `pin_PD13` and `pin_PF7` sorry, will send a fix for it.

Reply:
https://github.com/micropython/micropython/pull/12705